### PR TITLE
Intake arm

### PR DESCRIPTION
**Pull request by mlists**
Add logic to raise and lower the intake

### CODE REVIEW

File: components/intake.py [hunk 1]
```diff
@@ -5,32 +5,88 @@
...
 
     def __init__(self) -> None:
         self.motor = TalonFX(TalonIds.intake)
+
+        self.deploy_motor = CANSparkMax(
+            SparkMaxIds.intake_deploy, CANSparkMax.MotorType.kBrushless
+        )
+        self.pid_controller = self.deploy_motor.getPIDController()
+        self.encoder = self.deploy_motor.getEncoder()
+        RPMTORotsPerSec = tau * 60
```
Comment (mlists) [inline components/intake.py:38]: This conversion looks wrong, I think this is supposed to be from RPM to rad/s

File: components/intake.py [hunk 2]
```diff
@@ -44,14 +100,33 @@ def is_note_present(self) -> bool:
...
+                self.enable_soft_limits(True)
+                self.climb_encoder.setPosition(self.LIMIT_TOP)
+
         intake_request = VoltageOut(self.direction.value * self.motor_speed * 12.0)
 
         self.motor.set_control(intake_request)
+
+        self.pid_controller.setReference(
+            self.deploy_dir, CANSparkMax.ControlType.kSmartMotion
+        )
```
Comment (mlists) [inline components/intake.py:131]: This should be a position not a direction

**Comment by mlists** [comment]
We also have a lot of variables with incorrect names, see mypy

File: components/intake.py [hunk 3]
```diff
@@ -5,32 +5,94 @@
...
+
+from rev import CANSparkMax
+
+from ids import TalonIds, SparkMaxIds, DioChannels
 
 
 class IntakeComponent:
     motor_speed = tunable(0.4)
 
+    GEAR_RATIO = 24 / 48  # 24 / 54 / 48
```
Comment (auscompgeek) [inline components/intake.py:20]: ?

**Review by fotnitfpro000 — APPROVED** [review]
looks good

**Pull request merged**